src/lib/modules/compiler/index.ts

```diff
@@ -27,15 +29,30 @@ class Compiler {
     };
 
     async.eachObject(this.getAvailableCompilers(),
-      (extension: string, compiler: any, next: any) => {
+      (extension: string, compilers: any, next: any) => {
         const matchingFiles = contractFiles.filter(this.filesMatchingExtension(extension));
         if (matchingFiles.length === 0) {
           return next();
         }
 
-        compiler.call(compiler, matchingFiles, compilerOptions, (err: any, compileResult: any) => {
-          Object.assign(compiledObject, compileResult);
-          next(err, compileResult);
+        async.someLimit(compilers, 1, (compiler: any, someCb: Callback<boolean>) => {
+          compiler.call(compiler, matchingFiles, compilerOptions, (err: any, compileResult: any) => {
+            if (compileResult === false) {
```

Comment:
Should we also check for `err` here?
```suggestion
            if (err) { return someCb(err); }
            if (compileResult === false) {
```